--- evaluate.py.bak	2026-08-04 15:50:53.767500633 +0100
+++ evaluate.py	2026-08-04 15:50:53.779364904 +0100
@@ -61,7 +61,7 @@
     scores = []
     timeout_cnt = 0 
 
-    with ProcessPool(max_workers=1) as pool:
+    with ProcessPool(max_workers=16) as pool:
         future = pool.map(math_equal_process, params, timeout=3)
         iterator = future.result()
         with tqdm(total=len(samples), desc="Evaluate") as progress_bar:
